yt_dlp/extractor/navernow.py

```diff
@@ -0,0 +1,198 @@
+        vod_info = self._call_api(
+            f'/shows/{show_id}/vod/{replay_id}', replay_id,
+            note=f'Downloading JSON metadata for replay {show_id}-{replay_id}')
+        in_key = self._call_api(
+            f'/shows/{show_id}/vod/{replay_id}/inkey', replay_id,
+            note=f'Downloading JSON inkey for replay {show_id}-{replay_id}')['inKey']
+        return merge_dicts(
+            {
+                'id': f'{show_id}-{replay_id}',
+                'title': vod_info.get('episode', {}).get('title'),
+                'upload_date': unified_strdate(vod_info.get('episode', {}).get('start_time')),
+                'timestamp': unified_timestamp(vod_info.get('episode', {}).get('start_time')),
+                'thumbnail': vod_info.get('thumbnail_image_url'),
+            },
+            self._extract_video_info(replay_id, vod_info['video_id'], in_key))
+
+    def _extract_highlight(self, show_id, highlight_id, highlights=None):
+        page = 0
+        while True:
+            highlights_videos = highlights or self._call_api(
+                f'/shows/{show_id}/highlights/videos/', highlight_id,
+                query={'offset': page * self._PAGE_SIZE, 'limit': self._PAGE_SIZE},
+                note=f'Downloading JSON highlights for show {show_id} - page {page}')
+            highlight = [
+                v for v in highlights_videos.get('results', [])
+                if v.get('id', -1) == int(highlight_id)
+            ]
+            if highlight or highlights_videos.get('count', 0) <= self._PAGE_SIZE * (page + 1):
+                break
+            page += 1
+
+        if not highlight:
+            raise ExtractorError(f'Unable to find highlight {highlight_id} for show {show_id}')
+
+        highlight = highlight[0]
+        return merge_dicts(
+            {
+                'id': f'{show_id}-{highlight_id}',
+                'title': highlight.get('title'),
+                'upload_date': unified_strdate(highlight.get('regdate')),
+                'timestamp': unified_timestamp(highlight.get('regdate')),
+                'thumbnail': highlight.get('thumbnail_url'),
+            },
+            self._extract_video_info(
+                highlight_id, highlight['video_id'], highlight['video_inkey']))
+
+    def _extract_show_replays(self, show_id):
+        page = 0
+        entries = []
+        while True:
+            show_vod_info = self._call_api(
+                f'/vod-shows/{show_id}', show_id,
+                query={'offset': page * self._PAGE_SIZE, 'limit': self._PAGE_SIZE},
+                note=f'Downloading JSON vod list for show {show_id} - page {page}'
+            ).get('response', {}).get('result', {})
+            for v in show_vod_info.get('vod_list', []):
+                entries.append(self._extract_replay(show_id, v['id']))
+
+            if show_vod_info.get('count', 0) <= self._PAGE_SIZE * (page + 1):
+                break
+            page += 1
+        return entries
```

Comment:
Use a PagedList

Reply:
The merged case cannot be paged. But the individual cases can.

We can make it a pagedlist and then do this for the merged case:
```py
entries = itertools.chain(
    self._extract_show_replays(show_id), self._extract_show_highlights(show_id))
```